onnxruntime/test/providers/cpu/model_tests.cc

```diff
@@ -524,7 +548,31 @@ TEST_P(ModelTest, Run) {
       } else if (provider_name == "nuphar") {
         ASSERT_STATUS_OK(session_object.RegisterExecutionProvider(DefaultNupharExecutionProvider()));
       } else if (provider_name == "tensorrt") {
-        ASSERT_STATUS_OK(session_object.RegisterExecutionProvider(DefaultTensorrtExecutionProvider()));
+        if (test_case_name.find(ORT_TSTR("FLOAT16")) != std::string::npos) {
```

Comment:
I heard M60 doesn't have good support for FLOAT16. Would it be the reason why they are failing? Do you want to change the GPU to a different type?

Reply:
I think we're using m60 for TRT CI pipeline and T4 for TRT packaging pipeline? 
how do we make the tests pass regardless of which GPU we use? I think there are some api's which can tell us if fp16 is supported by the hardware. we can refine this in future PR?

Reply:
Sure.

Reply:
Yes, TensorRT has api to do this. If we want to detect whether fp16 is available for underline GPU device during test time, TRT EP needs to expose this api as well.